packages/flutter/test/rendering/object_test.dart

```diff
@@ -23,16 +26,33 @@ void main() {
   });
 
   test('detached RenderObject does not do semantics', () {
-    final TestRenderObject renderObject = new TestRenderObject();
+    final TestSemanticRenderObject renderObject = new TestSemanticRenderObject();
     expect(renderObject.attached, isFalse);
     expect(renderObject.describeSemanticsConfigurationCallCount, 0);
 
     renderObject.markNeedsSemanticsUpdate();
     expect(renderObject.describeSemanticsConfigurationCallCount, 0);
   });
+
+  test('RenderObject can capture images of itself', () async {
+    RenderRepaintBoundary boundary = new RenderRepaintBoundary();
+    layout(boundary, constraints: new BoxConstraints.tight(const Size(100.0, 200.0)));
+    pumpFrame(phase: EnginePhase.composite);
+    ui.Image image = await boundary.toImage();
+    expect(image.width, equals(100));
+    expect(image.height, equals(200));
+
+    // Now with pixel ratio set to something other than 1.0.
+    boundary = new RenderRepaintBoundary();
+    layout(boundary, constraints: new BoxConstraints.tight(const Size(100.0, 200.0)));
+    pumpFrame(phase: EnginePhase.composite);
+    image = await boundary.toImage(pixelRatio: 2.0);
+    expect(image.width, equals(200));
+    expect(image.height, equals(400));
+  });
```

Comment:
You should add a test that verifies that it works (gives the right pixels) with two layers, since that was a bug.

Reply:
Done.